ext_libs/ext_libs.cmake

```diff
@@ -118,3 +118,14 @@ else()
   # header at the root of the repo rather than its own nested sse2neon/ dir
   target_include_directories(sse2neon SYSTEM INTERFACE "${CMAKE_CURRENT_LIST_DIR}/sse2neon")
 endif()
+
+add_library(mlpack_ensmallen INTERFACE)
+# ensmallen can only be installed from source so doesn't really make sense to add it as a system dependency
+if(VW_ARMADILLO_SYS_DEP)
+  target_include_directories(mlpack_ensmallen INTERFACE ${CMAKE_CURRENT_LIST_DIR}/ensmallen/include)
+  find_package(Armadillo 9.800.0 REQUIRED)
+  target_include_directories(mlpack_ensmallen INTERFACE ${ARMADILLO_INCLUDE_DIRS})
```

Comment:
Please consume armadillo as a target instead of just referencing the include dirs

Reply:
done